plugins/client.go

```diff
@@ -0,0 +1,217 @@
+package plugins
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/alibaba/pouch/client"
+
+	"github.com/sirupsen/logrus"
+)
+
+var defaultTimeout = 30
+
+// defaultContentTypeis the default content type accepted and sent by the plugins.
+const defaultContentType = "application/vnd.docker.plugins.v1.1+json"
+
+// PluginClient the client which call Plugin ServiceMethod
+type PluginClient struct {
+	// address is the plugin server address
+	address string
+	// baseURL is the request base url
+	baseURL string
+	// the http client
+	client *http.Client
+}
+
+// NewPluginClient creates a PluginClient from the address and tls config
+func NewPluginClient(addr string, tlsconfig *TLSConfig) (*PluginClient, error) {
+	var config *tls.Config
+	var baseURL string
+	var err error
+
+	if addr == "" {
+		return nil, fmt.Errorf("empty plugin addr is invalid")
+	}
+
+	u, err := url.Parse(addr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse plugin addr: %s", addr)
+	}
+
+	baseURL = generateBaseURL(u)
+
+	// Scheme is https, generate the tls config
+	if u.Scheme == "https" {
+		if tlsconfig != nil && tlsconfig.InsecureSkipVerify == false {
+			config, err = client.GenTLSConfig(tlsconfig.KeyFile, tlsconfig.CertFile, tlsconfig.CAFile)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse plugin tls config")
```

Comment:
please throw out the error. I think that will make the error message more explicit.